tests/ProductBundle/Seo/Services/FacebookTest.php

```diff
@@ -71,4 +73,66 @@ public function testAlterPage()
 
         $this->assertContains('O-some product', $content);
     }
+
+    public function testAlterPageImage()
+    {
+        $mediaPool = $this->createMock(Pool::class);
+        $seoPage = new SeoPage('test');
+        $extension = new SeoExtension($seoPage, 'UTF-8');
+        $numberHelper = $this->createMock(NumberHelper::class);
+        $currencyDetector = $this->createMock(CurrencyDetectorInterface::class);
+        $product = new ProductFbMock();
+        $router = $this->createMock(RouterInterface::class);
+
+        //Prepare currency
+        $currency = new Currency();
+        $currency->setLabel('EUR');
+        $currencyDetector->expects($this->any())
+                ->method('getCurrency')
+                ->willReturn($currency);
+
+        $imageDummyLink = '/upload/dummy.png';
+        $domaine = 'http://my-domain.ltd';
+
+        // Test getImage
+        $image = $this->createMock(MediaInterface::class);
+
+        $imageProvider = $this->createMock(ImageProvider::class);
+        $imageProvider->expects($this->any())
+            ->method('generatePublicUrl')->willReturn($imageDummyLink);
+
+        $image->expects($this->any())
+            ->method('getName')->willReturn('correctMedia');
+        $image->expects($this->any())
+            ->method('getWidth')->willReturn(1111);
+        $image->expects($this->any())
+            ->method('getHeight')->willReturn(2222);
+        $image->expects($this->any())
+            ->method('getProviderName')->willReturn($imageProvider);
+        $image->expects($this->any())
+            ->method('getContentType')->willReturn('image/png');
+
+        $mediaPool->expects($this->any())
+            ->method('getProvider')->willReturn($imageProvider);
+
+        $product->setImage($image);
+
+        // Check if the header data are correctly registered
+        $fbService = new Facebook($router, $mediaPool, $numberHelper, $currencyDetector, $domaine, 'test', 'reference');
+        $fbService->alterPage($seoPage, $product);
+        $content = $extension->getHeadAttributes();
+
+        $this->assertContains('fb: http://ogp.me/ns/fb#', $content);
+
+        $content = $extension->getMetadatas();
+
+        // image link
+        $this->assertContains($domaine.$imageDummyLink, $content);
+        //image width
+        $this->assertContains('1111', $content);
```

Comment:
Please use a more specific assert. You are checking for the width, but the assert does not show this.

Reply:
Is it better now ?